dvc/output/base.py

```diff
@@ -287,6 +291,9 @@ def save(self):
             return
 
         self.hash_info = self.get_hash()
+        self.isexec = os.path.isfile(self.def_path) and os.access(
+            self.def_path, os.X_OK
+        )
```

Comment:
```suggestion
        self.isexec = self.tree.isfile(self.path_info) and self.tree.isexec(self.path_info)
```

Reply:
Good point! I've done this and also updated `utils:is_exec` function to return boolean by casting to `bool`:

https://github.com/iterative/dvc/blob/167ab79b8029d3272e6aa056c12305b6c9befeee/dvc/utils/__init__.py#L463